src/web/mjs/connectors/SmackJeeves.mjs

```diff
@@ -1,130 +1,99 @@
-     */
     constructor() {
         super();
         super.id = 'smackjeeves';
         super.label = 'Smack Jeeves';
         this.tags = [ 'webtoon', 'english' ];
-        this.url = 'http://www.smackjeeves.com';
+        this.url = 'https://www.smackjeeves.com';
+        this.he = require('he');
     }
 
-    /**
-     *
-     */
-    _getMangaListFromPages( mangaPageLinks, index ) {
-        index = index || 0;
-        return this.fetchDOM( mangaPageLinks[ index ], 'div.comic-container a.card', 5 )
-            .then( data => {
-                let mangaList = data.map( element => {
-                    return {
-                        id: this.getRelativeLink( element ),
-                        title: element.querySelector( 'div.title' ).textContent.trim()
-                    };
-                } );
-                if( index < mangaPageLinks.length - 1 ) {
-                    return this._getMangaListFromPages( mangaPageLinks, index + 1 )
-                        .then( mangas => mangas.concat( mangaList ) );
-                } else {
-                    return Promise.resolve( mangaList );
+    async _getCategories( url) {
+        const script = `
+            new Promise((resolve, reject) => {
+                try {
+                    resolve(cmnData);
+                } catch(error) {
+                    reject(error);
                 }
-            } );
+            });
+        `;
+        const request = new Request(new URL(url), this.requestOptions);
+        const data = await Engine.Request.fetchUI(request, script);
+
+        return data.navigation.items;
     }
 
-    /**
-     *
-     */
-    _getMangaList( callback ) {
-        let path = '/search.php?last_update=6&sort_by=4';
-        this.fetchDOM( this.url + path, 'div.comic-nav div#browse-page-select select option' )
-            .then( data => {
-                let pageLinks = data.map( option => this.url + path + '&start=' + option.value );
-                return this._getMangaListFromPages( pageLinks );
-            } )
-            .then( data => {
-                callback( null, data );
-            } )
-            .catch( error => {
-                console.error( error, this );
-                callback( error, undefined );
-            } );
+    async _getMangas() {
+        const categories = await this._getCategories( new URL('/discover?type=genre', this.url));
+
+        let manga_list = [];
+        for (const category of categories) {
+            let url = new URL('/api/getTitleListByGenreDiscover', this.url);
+            url.searchParams.set('genre', category.val);
+            url.searchParams.set('order', 'newarriva');
+
+            let pages = 1;
+            for (let page = 1; page <= pages; page++) {
+                url.searchParams.set('page', page);
+                const data = await this._fetchPOST(url);
+                pages = data.result.totalPageCnt;
+
+                manga_list.push(...data.result.list.map(element => {
```

Comment:
May use e.g. `item` instead of `element`, since the latter may make developers believe this is an HTML element (which wouldn't require to deal with HTML entities)

Reply:
What's the idea here? Is it better practice or easier readable for others?

Reply:
Readability, i was confused at first because the loop seemed to run only once and then noted that the `pages` variable in the condition of the for loop got changed dynamically

Reply:
Readability:
```javascript
let bike = new Car();
let car = new Car();
```